### ISSUE TITLE
feat: integrate NVIDIA DALI for GPU-accelerated augmentation

### ISSUE THREAD

**Issue by aymuos15** [aymuos15/nnBenchmark#31]
## Summary

Replace MONAI CacheDataset + PyTorch DataLoader with NVIDIA DALI for the online augmentation stage. DALI's compiled C++ execution engine uses separate CUDA stream pools for augmentation, enabling true prefetching overlap with training.

## Why

Current pipeline: CPU workers augment → queue → GPU trains. Augmentation and training overlap via multiprocessing, but augmentation is CPU-bound.

MONAI's GPU augmentation (EnsureTyped + ThreadDataLoader) doesn't help because both augmentation and training submit to the same CUDA stream — they serialize.

DALI solves this with its own CUDA stream pool + queue-based prefetching. Proven 2x speedup on 3D medical segmentation (MLPerf KITS19 UNet3D).

## Architecture

```
Current:  [CPU workers: augment] → queue → [GPU: train]
DALI:     [DALI GPU stream: augment] → prefetch queue → [PyTorch GPU stream: train]
```

## Scope

- Keep offline preprocessing as-is (crop to nonzero, save to disk)
- Replace online augmentation + data loading with DALI pipeline
- DALI loads preprocessed files, runs spatial/intensity augmentation on GPU
- Training loop receives PyTorch CUDA tensors via DALIGenericIterator (zero-copy)

## Key DALI operators for nnU-Net-style pipeline

- `fn.readers.numpy` — load preprocessed volumes
- `fn.segmentation.random_object_bbox` — foreground oversampling on GPU
- `fn.warp_affine` / `fn.rotate` — spatial augmentation
- `fn.flip` — mirroring
- `fn.gaussian_blur` — blur augmentation
- `fn.brightness_contrast` — intensity augmentation
- `fn.normalize` — z-score normalization

## References

- https://developer.nvidia.com/blog/accelerating-medical-image-processing-with-dali/
- https://docs.nvidia.com/deeplearning/dali/user-guide/docs/pipeline.html
- https://github.com/mlcommons/training/tree/master/image_segmentation/pytorch
